test/jdk/jdk/incubator/vector/templates/X-LoadStoreTest.java.template

```diff
@@ -702,18 +722,157 @@ public class $vectorteststype$ extends AbstractVectorLoadStoreTest {
+           }
+       }
+
+       int index = fi.apply(a.length);
+       boolean shouldFail = isIndexOutOfBounds(SPECIES.length(), index, a.length);
+       try {
+           VectorShuffle<$Boxtype$> shuffle = shuffleFromArray(a, index);
+           shuffleIntoArray(shuffle, r, index);
+           if (shouldFail) {
+               Assert.fail("Failed to throw IndexOutOfBoundsException");
+           }
+       } catch (IndexOutOfBoundsException e) {
+           if (!shouldFail) {
+               Assert.fail("Unexpected IndexOutOfBoundsException");
+           }
+       }
+   }
+
+   @Test(dataProvider = "shuffleIntProviderForIOOBE")
+   static void loadShuffleArrayIOOBE(IntFunction<int[]> fa, IntFunction<Integer> fi) {
+       int[] a = fa.apply(SPECIES.length());
+       int[] r = new int[a.length];
+
+       for (int ic = 0; ic < INVOC_COUNT; ic++) {
+           for (int i = 0; i < a.length; i += SPECIES.length()) {
+               VectorShuffle<$Boxtype$> shuffle = shuffleFromArray(a, i);
+               shuffle.intoArray(r, i);
+           }
+       }
+
+       int index = fi.apply(a.length);
+       boolean shouldFail = isIndexOutOfBounds(SPECIES.length(), index, a.length);
+       try {
+           shuffleFromArray(a, index);
+           if (shouldFail) {
+               Assert.fail("Failed to throw IndexOutOfBoundsException");
+           }
+       } catch (IndexOutOfBoundsException e) {
+           if (!shouldFail) {
+               Assert.fail("Unexpected IndexOutOfBoundsException");
+           }
+       }
+   }
+
+   @Test(dataProvider = "shuffleIntMemorySegmentProvider")
+   static void loadStoreShuffleMemorySegment(IntFunction<int[]> fa,
+                                      IntFunction<MemorySegment> fb,
+                                      ByteOrder bo) {
+       MemorySegment a = toShuffleSegment(SPECIES, fa.apply(SPECIES.length()), fb);
+       MemorySegment r = fb.apply((int) a.byteSize());
+
+       int l = (int) a.byteSize();
+       int s = SPECIES.length() * 4; //An integer for every lane is read out. So 4 bytes per lane
+
+       for (int ic = 0; ic < INVOC_COUNT; ic++) {
+           for (int i = 0; i < l; i += s) {
+               VectorShuffle<$Boxtype$> shuffle = VectorShuffle.fromMemorySegment(SPECIES, a, i, bo);
+               shuffle.intoMemorySegment(r, i, bo);
+           }
+       }
+       for (int i = 0; i < l; i += s) {
+           VectorShuffle<$Boxtype$> shuffle = VectorShuffle.fromMemorySegment(SPECIES, r, i, bo);
+           Assert.assertEquals(shuffle, VectorShuffle.fromMemorySegment(SPECIES, a, i, bo));
+       }
+   }
+
+   @Test(dataProvider = "shuffleIntByteProviderForIOOBE")
+   static void shuffleLoadMemorySegmentIOOBE(IntFunction<int[]> fa, IntFunction<Integer> fi) {
+       MemorySegment a = toShuffleSegment(SPECIES, fa.apply(SPECIES.length()), i -> Arena.ofAuto().allocate(i, Integer.SIZE));
+       MemorySegment r = Arena.ofAuto().allocate(a.byteSize(), $Boxtype$.SIZE);
```

Comment:
The alignment argument is not required here and is a little suspect because it aligns to the size of the vector element. I think we can avoid using it. I notice the memory segment vector access tests also use alignment, but it should not be needed either. I think this might have been required when the API was in preview.